src/api/spec/components/previews/status_message_component_preview.rb

```diff
@@ -0,0 +1,14 @@
+class StatusMessageComponentPreview < ViewComponent::Preview
+  # Preview at http://HOST:PORT/rails/view_components/status_message_component/as_anonymous_user
+  def as_anonymous_user
+    render(StatusMessageComponent.new(status_message: StatusMessage.first))
+  end
+
+  # Preview at http://HOST:PORT/rails/view_components/status_message_component/as_admin_user
+  def as_admin_user
+    # TODO: run_as doesn't change anything somehow... it's still the anonymous user in the session when rendering the view component (so the delete icon is not displayed)
```

Comment:
I really tried to find out why this isn't working, but I couldn't figure out why. I don't see this as a blocker to merge this PR, although it would be nice to have view component previews for various roles.